packages/launcher/lib/browsers.ts

```diff
@@ -1,14 +1,161 @@
 import Debug from 'debug'
 import * as cp from 'child_process'
-import { browsers, FoundBrowser } from '@packages/types'
+import type { Browser, BrowserValidatorResult, FoundBrowser } from '@packages/types'
 import type { Readable } from 'stream'
 
 export const debug = Debug('cypress:launcher:browsers')
 
-export { browsers }
+// Chrome started exposing CDP 1.3 in 64
+export const MIN_CHROME_VERSION = 64
+
+// Firefox started exposing CDP in 86
+export const MIN_FIREFOX_VERSION = 86
+
+// Edge switched to Blink in 79
+export const MIN_EDGE_VERSION = 79
+
+// Compares a detected browser's major version to its minimum supported version
+// to determine if the browser is supported by Cypress.
+export const validateMinVersion = (browser: FoundBrowser): BrowserValidatorResult => {
+  const minSupportedVersion = browser.minSupportedVersion
+  const majorVersion = browser.majorVersion
+
+  if (majorVersion && minSupportedVersion && parseInt(majorVersion) < minSupportedVersion) {
+    return {
+      isSupported: false,
+      warningMessage: `Cypress does not support running ${browser.displayName} version ${majorVersion}. To use ${browser.displayName} with Cypress, install a version of ${browser.displayName} newer than or equal to ${minSupportedVersion}.`,
+    }
+  }
+
+  return {
+    isSupported: true,
+  }
+}
 
 /** list of the browsers we can detect and use by default */
 
+export const browsers: Browser[] = [
```

Comment:
With the additional logic in play here, it didn't feel right keeping the default browsers defined within the `types` package. It was a little weird to have them there in the first place?

Reply:
:100: